network-area-diagram/src/main/java/com/powsybl/nad/svg/SvgWriter.java

```diff
@@ -129,13 +129,57 @@ private void drawBranchEdges(Graph graph, XMLStreamWriter writer) throws XMLStre
 
             drawHalfEdge(graph, writer, edge, BranchEdge.Side.ONE);
             drawHalfEdge(graph, writer, edge, BranchEdge.Side.TWO);
+
+            if (BranchEdge.PST_EDGE.equals(edge.getType())) {
+                drawPstArrow(writer, edge);
+            }
+            if (edge.getType().equals(BranchEdge.HVDC_LINE_EDGE)) {
+                drawConverterStation(writer, edge);
+            }
+            if (BranchEdge.LINE_EDGE.equals(edge.getType()) && svgParameters.isLineNameDisplayed()) {
+                drawLineLabel(writer, edge);
+            }
+
             drawEdgeCenter(writer, edge);
 
             writer.writeEndElement();
         }
         writer.writeEndElement();
     }
 
+    private void drawLineLabel(XMLStreamWriter writer, BranchEdge edge) throws XMLStreamException {
+
+        String lineName = edge.getEquipmentId();
+
+        if (lineName != null) {
+
+            double edgeAngle = edge.getEdgeEndAngle(BranchEdge.Side.ONE);
+            boolean textFlipped = Math.cos(edgeAngle) < 0;
+            double textAngle = textFlipped ? edgeAngle - Math.PI : edgeAngle;
+
+            List<Point> points1 = edge.getPoints1();
+            List<Point> points2 = edge.getPoints2();
+            Point anchorPoint = Point.createMiddlePoint(points1.get(points1.size() - 1), points2.get(points2.size() - 1));
+            String style = "text-anchor:middle";
+
+            if (!edge.isVisible(BranchEdge.Side.ONE)) {
+                edgeAngle = edge.getEdgeEndAngle(BranchEdge.Side.TWO);
+                textFlipped = Math.cos(edgeAngle) < 0;
+                style = textFlipped ? "text-anchor:end" : "text-anchor:start";
+            } else if (!edge.isVisible(BranchEdge.Side.TWO)) {
+                style = textFlipped ? "text-anchor:end" : "text-anchor:start";
+            }
+
+            writer.writeStartElement(GROUP_ELEMENT_NAME);
+            writer.writeAttribute(CLASS_ATTRIBUTE, StyleProvider.EDGE_ID_CLASS);
```

Comment:
Add the `GLUED_CENTER_CLASS` at the same time, using `writeStyleClasses` method.

Reply:
In order to take into account another of your points, I would like to call the `drawLineLabel(...)` inside the `drawEdgeCenter(...)` function, where there is already a `writer.writeAttribute(CLASS_ATTRIBUTE, StyleProvider.GLUED_CENTER_CLASS)`.
I was thinking of not adding the GLUED_CENTER_CLASS to the drawLineLabel(...) function. 
What do you think about that ?